core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -245,70 +253,86 @@ export class RteHelperModalComponent {
-    >[];
-    let url: string = tmpCustomizationArgs[0].value;
-    let text: string = tmpCustomizationArgs[1].value;
+  updateRteSaveButtonState(value: number | string | boolean): void {
+    if (this.componentId === this.COMPONENT_ID_MATH) {
+      let rawLatex: string = value[0].raw_latex;
+      let mathExpressionSvgIsBeingProcessed: boolean = (
+        value[0].mathExpressionSvgIsBeingProcessed);
+      this.saveButtonIsDisabled = (
+        mathExpressionSvgIsBeingProcessed || rawLatex === '');
+    } else if (this.componentId === this.COMPONENT_ID_VIDEO) {
+      let start: number = value[1];
+      let end: number = value[2];
+      if (start === 0 && end === 0) {
+        this.saveButtonIsDisabled = false;
+      } else {
+        this.saveButtonIsDisabled = start >= end;
+      }
+      this.updateRteErrorMessage();
+    } else if (this.componentId === this.COMPONENT_ID_LINK) {
+      let url: string = value[0];
+      let text: string = value[1];
+      // First check if the `text` looks like a URL.
+      const suffixes = ['.com', '.org', '.edu', '.gov'];
+      let urlSuffixIsValid = false;
 
-    // First check if the `text` looks like a URL.
-    const suffixes = ['.com', '.org', '.edu', '.gov'];
-    let textLooksLikeUrl = false;
-    for (const suffix of suffixes) {
-      if (text.endsWith(suffix)) {
-        textLooksLikeUrl = true;
+      for (const suffix of suffixes) {
+        if (url !== '' && url.endsWith(suffix)) {
+          urlSuffixIsValid = true;
+          break;
+        } else {
+          urlSuffixIsValid = false;
+          this.saveButtonIsDisabled = true;
+        }
       }
-    }
-    if (!textLooksLikeUrl) {
-      return false;
-    }
-    // If the text looks like a URL, strip the leading 'http://' or
-    // 'https://' or 'www.'.
-    const prefixes = ['https://', 'http://', 'www.'];
-    for (const prefix of prefixes) {
-      if (url.startsWith(prefix)) {
-        url = url.substring(prefix.length);
+      if (urlSuffixIsValid) {
+        // The link text. If left blank, the link URL will be used.
+        if (text === '') {
+          text = url;
+          this.saveButtonIsDisabled = false;
+        }
+        // If the text looks like a URL, strip the leading 'http://' or
+        // 'https://' or 'www.'.
+        const prefixes = ['https://', 'http://', 'www.'];
+        for (const prefix of prefixes) {
+          if (url.startsWith(prefix)) {
+            url = url.substring(prefix.length);
+          }
+          if (text.startsWith(prefix)) {
+            text = text.substring(prefix.length);
+          }
+        }
+        // After the cleanup, if the strings are not equal, then we do not
+        // allow the lesson creator to save it.
+        this.saveButtonIsDisabled = url !== text;
+        this.updateRteErrorMessage();
       }
-      if (text.startsWith(prefix)) {
-        text = text.substring(prefix.length);
+    }
+  }
+
+  updateRteErrorMessage(): void {
```

Comment:
This stuff shouldn't be separate, it should be integrated with the function above. When an error is discovered then maybe call a helper function that updates the error message (which is passed in to the helper function) and sets the save button to disabled.

Otherwise you can't distinguish between different types of errors for the same component. It also seems like good practice to always show an error if we forbid saving, so that the creator knows what they need to do to fix it.

Reply:
is it preferred to have a function like this
```ts
  updateRteErrorMessage(errorMessage: string): void {
    this.errorMessage = errorMessage;
  }
```
for readability or to just set 
```ts
if (this.saveButtonIsDisabled) {
        this.errorMessage = (
          'Please ensure that the start time of the video is earlier than ' +
          'the end time.');
      } else {
        this.errorMessage = '';
      }
```
directly without any helper function?

Reply:
done, made a helper function to set the message

Reply:
Either's fine ... though stylistically, I think what you did (the helper function) is better since it's got a clear contract and is reusable functionality :).